okhttp/src/main/java/okhttp3/Dns.kt

```diff
@@ -54,5 +54,10 @@ interface Dns {
         }
       }
     }
+
+    // This lambda conversion is for Kotlin callers expecting a Java SAM (single-abstract-method).
+    @JvmName("-deprecated_Dns")
+    inline operator fun invoke(crossinline block: (String) -> List<InetAddress>): Dns =
+        Dns { block(it) }
```

Comment:
An alternative we should consider – top-level function named with the type name:

```
inline fun Dns(crossinline block: (String) -> List<InetAddress>): Dns = object : Dns {
  override fun lookup(hostname: String) = block(hostname)
}
```

Reply:
As also found here:
https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/Job.kt#L357